engine/src/tangl/utils/base_model_plus.py

```
@@ -40,6 +41,130 @@ def _convert_sets_to_list(self, nxt_ser):
+
+        # If disambiguated, return the first hit (MRO-ordered).
+        if generic_cls is not None:
+            return results[0]
+
+        # Otherwise ensure uniqueness.
+        uniq = list(dict.fromkeys(results))
+        if len(uniq) != 1:
+            raise ValueError(
+                f"Ambiguous generic bases for {cls.__name__}; "
+                f"pass generic_cls=... to disambiguate. Found: {uniq}"
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)